Assignment/Node.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+namespace GenericsHomework;
+
+
+public class Node<T> : IEnumerable<Node<T>> where T : notnull
+{
+
+    public Node<T> Next { get; private set; }
+    public T Data { get; }
+
+    public Node(T data)
+    {
+        Data = data;
+        Next = this;
+    }
+
+    public void Append(T value)
+    {
+        if(Exists(value))
+        {
+            throw new InvalidOperationException(nameof(value));
+        }   
+        Node<T> newNode = new(value);
+        Node<T> cur = this;
```

Comment:
I know this is an old assignment, but would be good to rename cur to currentNode
```suggestion
        Node<T> currentNode = this;
```

Reply:
changed the name, thank you!